internal/services/batch/batch_pool_resource.go

```diff
@@ -159,15 +160,22 @@ func resourceBatchPool() *pluginsdk.Resource {
 										ForceNew:     true,
 										ValidateFunc: validation.StringIsNotEmpty,
 									},
+									"identity_reference": {
+										Type:         pluginsdk.TypeString,
+										Optional:     true,
+										ForceNew:     true,
+										ValidateFunc: msivalidate.UserAssignedIdentityID,
+										Description:  "The User Assigned Identity to use for Container Registry access.",
+									},
 									"user_name": {
 										Type:         pluginsdk.TypeString,
-										Required:     true,
+										Optional:     true,
 										ForceNew:     true,
 										ValidateFunc: validation.StringIsNotEmpty,
 									},
 									"password": {
 										Type:         pluginsdk.TypeString,
-										Required:     true,
+										Optional:     true,
 										ForceNew:     true,
```

Comment:
and I presume this is required when user is so
```suggestion
										ForceNew:     true,
										RequiredWith: []{"user"},
```

Reply:
Here is the existing configuration, exactly with the idea as mine: https://github.com/hashicorp/terraform-provider-azurerm/blob/main/internal/services/containers/container_registry_task_resource.go#L532
(Have not seen it before).